task004/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Shape interface {
+	String() string
+	Area() float64
+	Perimeter() float64
+}
+
+type Circle struct {
```

Comment:
Exported type should contain exportable fields or have a constructor. 
Currently, this type cannot be used outside the package.

https://play.golang.org/p/8OMazlGxikF